cpp/src/c_api/graph_mg.cpp

```diff
@@ -298,87 +297,178 @@ extern "C" cugraph_error_code_t cugraph_mg_graph_create(
-               *error);
+  size_t local_num_edges{0};
 
-  CAPI_EXPECTS((weights == nullptr) || (p_weights->size_ == p_src->size_),
-               CUGRAPH_INVALID_INPUT,
-               "Invalid input arguments: src size != weights size.",
-               *error);
+  cugraph_data_type_id_t vertex_type{cugraph_data_type_id_t::NTYPES};
+  cugraph_data_type_id_t weight_type{cugraph_data_type_id_t::NTYPES};
+
+  for (size_t i = 0; i < num_arrays; ++i) {
+    CAPI_EXPECTS(p_src[i]->size_ == p_dst[i]->size_,
+                 CUGRAPH_INVALID_INPUT,
+                 "Invalid input arguments: src size != dst size.",
+                 *error);
+
+    CAPI_EXPECTS(p_src[i]->type_ == p_dst[i]->type_,
+                 CUGRAPH_INVALID_INPUT,
+                 "Invalid input arguments: src type != dst type.",
+                 *error);
+
+    CAPI_EXPECTS((p_vertices == nullptr) || (p_src[i]->type_ == p_vertices[i]->type_),
+                 CUGRAPH_INVALID_INPUT,
+                 "Invalid input arguments: src type != vertices type.",
+                 *error);
+
+    CAPI_EXPECTS((weights == nullptr) || (p_weights[i]->size_ == p_src[i]->size_),
+                 CUGRAPH_INVALID_INPUT,
+                 "Invalid input arguments: src size != weights size.",
+                 *error);
+
+    local_num_edges += p_src[i]->size_;
+
+    if (vertex_type == cugraph_data_type_id_t::NTYPES) vertex_type = p_src[i]->type_;
+
+    if (weights != nullptr) {
+      if (weight_type == cugraph_data_type_id_t::NTYPES) weight_type = p_weights[i]->type_;
+    }
+
+    CAPI_EXPECTS(p_src[i]->type_ == vertex_type,
+                 CUGRAPH_INVALID_INPUT,
+                 "Invalid input arguments: all vertex types must match",
+                 *error);
+
+    CAPI_EXPECTS(p_weights[i]->type_ == weight_type,
+                 CUGRAPH_INVALID_INPUT,
+                 "Invalid input arguments: all weight types must match",
+                 *error);
+  }
+
+  size_t num_edges = cugraph::host_scalar_allreduce(p_handle->handle_->get_comms(),
+                                                    local_num_edges,
+                                                    raft::comms::op_t::SUM,
+                                                    p_handle->handle_->get_stream());
+
+  // FIXME:  Need to handle the case where a GPU gets a NULL pointer but other GPUs
+  //   Get values.  Override vertex_type/weight_type/edge_id_type but don't
+  //   fail.
+
+  auto vertex_types = cugraph::host_scalar_allgather(
+    p_handle->handle_->get_comms(), static_cast<int>(vertex_type), p_handle->handle_->get_stream());
+
+  auto weight_types = cugraph::host_scalar_allgather(
+    p_handle->handle_->get_comms(), static_cast<int>(weight_type), p_handle->handle_->get_stream());
+
+  CAPI_EXPECTS(
+    std::count_if(vertex_types.begin(),
+                  vertex_types.end(),
+                  [vertex_type](auto t) { return vertex_type != static_cast<int>(t); }) == 0,
```

Comment:
A minor point, but is `std::all_of` more specific for this? (https://en.cppreference.com/w/cpp/algorithm/all_any_none_of). I guess all_of may better document the intention.

Reply:
Will address when I get to the FIXME, which might change this logic anyway.

Reply:
Fixed in next push.